src/main.cpp

```diff
@@ -67,7 +321,7 @@ int main(int argc, const char* argv[]) {
     }
 #endif
 
-    if (configFileName.empty()) {
+    if (configFileName.empty() && config.Argmap.empty()) {
```

Comment:
```suggestion
    if (configFileName.empty()) {
```
You are forcing to use the configuration via command line options only when there is a configuration file.